lib/ldap.c

```diff
@@ -182,6 +182,41 @@ const struct Curl_handler Curl_handler_ldaps = {
 };
 #endif
 
+#ifdef USE_WIN32_LDAP
+static int Win_ldap_bind(struct connectdata *conn, LDAP *server, TCHAR *user, TCHAR *passwd)
+{
+    int rc = 0;
+    ULONG method = LDAP_AUTH_NEGOTIATE;
+    SEC_WINNT_AUTH_IDENTITY cred = { 0 };
+    PCHAR inuser = NULL;
+    PCHAR inpass = NULL;
+
+    if (user != NULL && passwd != NULL) {
+        cred.User = user;
+        cred.UserLength = strlen(user);
+        cred.Password = passwd;
+        cred.PasswordLength = strlen(passwd);
+        cred.Flags = SEC_WINNT_AUTH_IDENTITY_ANSI;
```

Comment:
> How is the behavior different from Linux with LDAP after this change?
Linux behavior doesn't affected with this change, but Windows curl now able to request Microsoft AD. Basic AUTH almost always disabled on Windows, so now without user/password curl will perform auto-negotiation. ```curl --ntlm -u user:pass ldap://....``` will connect with other user credentials.